exercises/protein-translation/example.rs

```diff
@@ -11,20 +11,21 @@ pub fn parse<'a>(pairs: Vec<(&'a str, &'a str)>) -> CodonInfo<'a> {
 }
 
 impl<'a> CodonInfo<'a> {
-    pub fn name_for(&self, codon: &str) -> Result<&'a str, &'static str> {
-        match self.actual_codons.get(&codon) {
-            Some(name) => Ok(name),
-            None => Err("Invalid"),
+    pub fn name_for(&self, codon: &str) -> Option<&'a str> {
+        if let Some(acid) = self.actual_codons.get(&codon) {
```

Comment:
This part feels really wrong (get `Some` to return `Some`), but if I used only `self.actual_codons.get(&codon)`,
the compiler would complain that the `Option<&&str>` is returned instead of `Option<&str>`. 
Any other way to bypass the problem?

Reply:
My own instinct would be to use `self.actual_codons.get(&codon).map(|&c| c)`, but there are a few other ways to handle it as well.

Reply:
The example should simply compile and pass tests,
so perhaps there is no need for more advanced concepts 
and it would be enough if I changed `if let` with `map`?

Apart from that are there any more changes you would request?

Reply:
No, that should be the last change, I think.